tests/test_schedule/test_request_handler.py

```diff
@@ -0,0 +1,177 @@
+        self.assertEqual(test_vtw_list, actual_vtw_list)
+
+    def test_number_of_scheduled_items(self):
+        self.request_handler.schedule_request(0, 1, 2, 3)
+        self.request_handler.schedule_request(1, 1, 2, 3)
+        self.assertEqual(self.request_handler.number_of_scheduled_requests, 2)
+
+    def test_sort_by_decreasing_priority(self):
+        self.request_handler.sort_by_decreasing_priority()
+
+        self.assertEqual(self.request_handler[0][RequestIndices.priority], 7)
+        self.assertEqual(self.request_handler[1][RequestIndices.priority], 7)
+        self.assertEqual(self.request_handler[2][RequestIndices.priority], 5)
+        self.assertEqual(self.request_handler[3][RequestIndices.priority], 4)
+        self.assertEqual(self.request_handler[4][RequestIndices.priority], 4)
+        self.assertEqual(self.request_handler[5][RequestIndices.priority], 3)
+        self.assertEqual(self.request_handler[6][RequestIndices.priority], 2)
+        self.assertEqual(self.request_handler[7][RequestIndices.priority], 1)
+        self.assertEqual(self.request_handler[8][RequestIndices.priority], 1)
+        self.assertEqual(self.request_handler[9][RequestIndices.priority], 1)
+
+    def test_sort_by_decreasing_opportunity(self):
+        self.request_handler.sort_by_decreasing_opportunity()
+
+        self.assertEqual(len(self.request_handler[0][RequestIndices.vtw_list]), 20)
+        self.assertEqual(len(self.request_handler[1][RequestIndices.vtw_list]), 19)
+        self.assertEqual(len(self.request_handler[2][RequestIndices.vtw_list]), 19)
+        self.assertEqual(len(self.request_handler[3][RequestIndices.vtw_list]), 19)
+        self.assertEqual(len(self.request_handler[4][RequestIndices.vtw_list]), 19)
+        self.assertEqual(len(self.request_handler[5][RequestIndices.vtw_list]), 19)
+        self.assertEqual(len(self.request_handler[6][RequestIndices.vtw_list]), 18)
+        self.assertEqual(len(self.request_handler[7][RequestIndices.vtw_list]), 18)
+        self.assertEqual(len(self.request_handler[8][RequestIndices.vtw_list]), 18)
+        self.assertEqual(len(self.request_handler[9][RequestIndices.vtw_list]), 18)
+
+    def test_sort_vtws_by_increasing_conflict_degree(self):
+        self.request_handler.sort_vtws_by_increasing_conflict_degree()
+        for request in self.request_handler:
+            previous_conflict_degree = None
+            for vtw in request[RequestIndices.vtw_list]:
+                current_conflict_degree = self.request_handler._conflict_degree(vtw)
+                if previous_conflict_degree is not None:
+                    self.assertLessEqual(previous_conflict_degree, current_conflict_degree)
+                previous_conflict_degree = current_conflict_degree
+
+    def test_sort_vtws_by_increasing_rise_time(self):
+        self.request_handler.sort_vtws_by_increasing_rise_time()
+        for request in self.request_handler:
+            previous_rise_time = None
```

Comment:
Setting `previous_rise_time = 0` should allow us to remove the `if previous_rise_time is not None` block and make the code cleaner. Setting to zero should not pose a problem.